code/clue.js

```diff
@@ -6,54 +6,271 @@ const mrGreen = {
+  firstName: 'Scarlet',
+  lastName: 'Cassandra',
+  color: 'red',
+  description: 'She has curly hair and a sharp look',
+  age: 37,
+  image: './assets/scarlet.png',
+  occupation: 'Coder',
+  favouriteWeapon: 'trophy'
+}
+
+const mrMustard = {
+  firstName: 'Jack',
+  lastName: 'Mustard',
+  color: 'orange',
+  description: 'He has been expelled from the army',
+  age: 58,
+  image: './assets/mustard.png',
+  occupation: 'colonel',
+  favouriteWeapon: 'pistol'
+
+}
+
+const mrsPeacock = {
+  firstName: 'Eleanor',
+  lastName: 'Peacock',
+  color: 'blue',
+  description: 'Her family used to be very rich',
+  age: 50,
+  image: './assets/peacock.png',
+  occupation: 'housewife',
+  favouriteWeapon: 'poison'
+
+}
+
+const mrsWhite = {
+  firstName: 'Sophia',
+  lastName: 'White',
+  color: 'yellow',
+  description: 'A lonely women, no family around',
+  age: 60,
+  image: './assets/white.png',
+  occupation: 'maid',
+  favouriteWeapon: 'rope'
+}
 
 const rope = {
   name: 'rope',
-  weight: 10
+  id: 'rope',
+  weight: 10,
+  location: 'Patio'
+}
+
+const knife = {
+  name: 'knife',
+  id: 'knife',
+  weight: 8,
+  location: 'Kitchen'
+}
+
+const candlestick = {
+  name: 'candlestick',
+  id: 'candleStick',
+  weight: 15,
+  location: 'Living Room'
+}
+
+const dumbbell = {
+  name: 'dumbbell',
+  id: 'dumbBell',
+  weight: 12,
+  location: 'Billiard Room'
+}
+
+const poison = {
+  name: 'poison',
+  id: 'poison',
+  weight: 1.5,
+  location: 'Conservatory'
+}
+
+const axe = {
+  name: 'axe',
+  id: 'axe',
+  weight: 8,
+  location: 'Patio'
 }
 
-// THE ROOMS ONLY HAS A NAME SO NO NEED FOR OBJECTS THERE.
+const bat = {
+  name: 'bat',
+  id: 'bat',
+  weight: 18,
+  location: 'Hall'
+}
 
-// NOW GROUP ALL SUSPECTS, WEAPONS AND ROOMS IN ARRAYS LIKE THIS:
+const trophy = {
+  name: 'trophy',
+  id: 'trophy',
+  weight: 17,
+  location: 'Lounge'
+}
+
+const pistol = {
+  name: 'pistol',
+  id: 'pistol',
+  weight: 1,
+  location: 'Study'
+}
 
 const suspects = [
   mrGreen,
+  mrMustard,
+  mrsPeacock,
+  mrPlum,
+  msCassandra,
   mrsWhite
-  // ...  and the rest
 ]
 
-const weapons = []
+const weapons = [
+  knife,
+  candlestick,
+  dumbbell,
+  poison,
+  axe,
+  bat,
+  trophy,
+  pistol
+]
 
-const rooms = []
+const rooms = [
+  'Dining Room', 
+  'Conservatory', 
+  'Kitchen',
+  'Study', 
+  'Library', 
+  'Billiard Room', 
+  'Lounge',
+  'Ballroom', 
+  'Hall', 
+  'Spa', 
+  'Living Room', 
+  'Observatory', 
+  'Theater', 
+  'Guest House', 
+  'Patio',
+]
 
-// THIS FUNCTION WILL RANDOMLY SELECT ONE ITEM FROM THE ARRAY THAT YOU PASS IN TO THE FUNCTION.
-// YOU DON'T NEED TO CHANGE THIS, JUST TRY TO UNDERSTAND IT. AND HOW TO USE IT.
-const randomSelector = array => {
+const randomSelector = (array) => {
   return array[Math.floor(Math.random() * array.length)]
 }
 
-// CREATE AN OBJECT THAT KEEPS THE MYSTERY.
-// With a killer, a weapon and a room.
-// The values will be set later.
+const mystery = {
+  killer: '',
+  weapon: '',
```

Comment:
It's even more semantic approach to initialise mystery object with `null` values like this :

```
const mystery = {
  killer: null,
  room: null,
  weapon: null
}
```

Moreover, it's great you used `const` keyword - we will be changing values of properties, that's true, but JavaScript will not consider this as overwriting. What counts for JavaScript is that object will stay object